File: little_brother/user_status.py
# -*- coding: utf-8 -*-

# Copyright (C) 2019-2021  Marcus Rickert
#
# See https://github.com/marcus67/little_brother_taskbar
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 3 of the License, or
# (at your option) any later version.
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# You should have received a copy of the GNU General Public License along
# with this program; if not, write to the Free Software Foundation, Inc.,
# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.


class UserStatus(object):

    def __init__(self, p_username=None):

        self.username = p_username
        self.minutes_left_in_session = None
        self.activity_allowed = False
        self.logged_in = False
        self.locale = None
        self.notification = None

        # Since LittleBrother version 0.3.13
        self.warning_time_without_send_events = None
        self.maximum_time_without_send_events = None

        # Since LittleBrother 0.4.1
        self.monitoring_active = False

        # Since LittleBrother 0.4.6
        self.optional_time_available = None
        self.ruleset_check_interval = None

File: littlebrother-frontend/src/app/models/user-status.ts
// Copyright (C) 2019-24  Marcus Rickert
//
// See https://github.com/marcus67/little_brother
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3 of the License, or
// (at your option) any later version.
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
// You should have received a copy of the GNU General Public License along
// with this program; if not, write to the Free Software Foundation, Inc.,
// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

import { UserStatusDetail } from '../models/user-status-detail'
import { get_duration_as_string, get_date_from_iso_string, get_date_as_string } from '../common/tools'

export class UserStatus {
  user_id?: number;
  full_name?: String;
	context_label?: String;
	todays_activity_duration_in_seconds?: number;
	max_time_per_day_in_seconds?: number;
	todays_downtime_in_seconds?: number;
	free_play?: boolean;
	activity_permitted?: boolean;
	previous_activity_start_time_in_iso_8601?: string;
	previous_activity_end_time_in_iso_8601?: string;
	current_activity_start_time_in_iso_8601?: string;
	current_activity_duration_in_seconds?: number;
	current_activity_downtime_in_seconds?: number;
	reasons?: string[];
  user_status_details?: UserStatusDetail[];

	constructor(otherObject?: object) {
	  if (otherObject)
	    Object.assign(this, otherObject)
	}

  todays_activity_duration() : string {
   return get_duration_as_string(this.todays_activity_duration_in_seconds);
  }

  todays_downtime() : string {
   return get_duration_as_string(this.todays_downtime_in_seconds);
  }

  max_time_per_day() : string {
   return get_duration_as_string(this.max_time_per_day_in_seconds);
  }

  current_activity_duration() : string {
   return get_duration_as_string(this.current_activity_duration_in_seconds);
  }

  current_activity_downtime() : string {
   return get_duration_as_string(this.current_activity_downtime_in_seconds);
  }

  current_activity_start_time() : Date {
    return get_date_from_iso_string(this.current_activity_start_time_in_iso_8601)
  }

  current_activity_start_time_as_string(): string {
    return get_date_as_string(this.current_activity_start_time());
  }

  previous_activity_start_time() : Date {
    return get_date_from_iso_string(this.previous_activity_start_time_in_iso_8601)
  }

  previous_activity_start_time_as_string(): string {
    return get_date_as_string(this.previous_activity_start_time());
  }

  previous_activity_end_time() : Date {
    return get_date_from_iso_string(this.previous_activity_end_time_in_iso_8601)
  }

  previous_activity_end_time_as_string(): string {
    return get_date_as_string(this.previous_activity_end_time());
  }
}
